Frameworks/UIKit/UITextField.mm

```diff
@@ -120,34 +113,30 @@ @implementation UITextField {
  @Status Interoperable
 */
 - (void)setText:(NSString*)text {
-    if (![_text isEqualToString:text]) {
-        if (text != nil) {
-            _text = [text copy];
-        } else {
-            _text = nil;
-        }
-
-        [_secureModeLock lock];
-        if (_secureTextMode) {
-            _passwordBox.password = _text;
-        } else {
-            _textBox.text = _text;
-            // Ensure caret at end of field in case we programmatically
-            // gain focus (becomeFirstResponder) after the text is set:
-            _textBox.selectionStart = [_text length];
-            _textBox.selectionLength = 0;
-        }
-        [_secureModeLock unlock];
-
-        [self _adjustFontSizeToFitWidthOrApplyCurrentFont];
+    [_secureModeLock lock];
+    if (_secureTextMode) {
+        _passwordBox.password = [text copy];
+    } else {
+        _textBox.text = [text copy];
+        // Ensure caret at end of field in case we programmatically
+        // gain focus (becomeFirstResponder) after the text is set:
+        _textBox.selectionStart = [text length];
+        _textBox.selectionLength = 0;
     }
+    [_secureModeLock unlock];
+
+    [self _adjustFontSizeToFitWidthOrApplyCurrentFont];
 }
 
 /**
  @Status Interoperable
 */
 - (NSString*)text {
```

Comment:
These are still going through manual verification through XAMLCatalog. we can add more functional test once the pattern is fixed. I believe Oliver is still working out the detail how functional test/XAMLCatalog can share viewcontroller code. Once the contract is defined. we should be able to start to author more automated tests. 

---
In reply to: [95400508](https://github.com/Microsoft/WinObjC/pull/1604#discussion_r95400508) [](ancestors = 95400508)